### DISCUSSION TITLE
Unicorn 🦄 to work along with ListView and Pagination

### DISCUSSION THREAD

**Discussion by chirdeeptomar** [django-commons/django-unicorn#194] [Q&A]
I am currently using Django's list view with Pagination, how should I go about using Unicorn view along with number paginator and update the results. Any ideas or samples?

**Comment by adamghill** [comment]
Pagination is something I've thought about a little bit, but haven't created an example for. I think it should be possible by using a queryset and Django paginator that gets set in a `hydrate` method, with something like `page_number = self.request.GET.get("page", 1)` to get the right "page". And then click events on the pagination numbers which update the paginator. The one part I don't handle currently is adding/updating a querystring (I have `HashUpdate` and `LocationUpdate`), but that might need to be added for everything to just work.

Thanks for starting the discussion! I'll try to create an example in the next few days to hopefully make that more clear. 👍

**Reply by algameko** [reply]
Hello,
I found this example [https://github.com/poiedk/django-unicorn-pagination-example/tree/main](url)
but I'm not able to make it working
I receive an error
`Cannot cache component '<class 't.components.lista_prodotti_t.ListaProdottiTView'>' because it is not picklable: <class 'TypeError'>: cannot pickle 'generator' object`
